docs/concepts/artifact.mdx

```diff
@@ -0,0 +1,239 @@
+
+### The Significance of Annotations
+
+Annotations in OCI artifacts offer a means of adding metadata to various
+components, including the Image Manifest, Image Index and Image Layers.
+These annotations hold key-value pairs that represent either the metadata
+of the OCI artifact or blobs, such as the creation time, or data of the artifact itself.
+
+```json
+{
+  ...
+  "annotations": {
+    "oci.opencontainers.image.created": "2023-01-02T03:04:05Z",
+    "com.example.information": "useful-info"
+  }
+}
+```
+
+### Determining the artifact type
+
+For OCI artifacts the **type** of the artifact may be determined from the `artifactType` property
+in the image manifest or index. This property was introduced in v1.1 of the [image specification][image-properties].
+
+In cases when the `artifactType` property is not defined, the `config.mediaType` property may be used to determine
+the type of the artifact as shown in the diagram below.
+
+```mermaid
+flowchart TD
+    B{Is\nmanifest.artifactType\ndefined?}
+    B -->|Yes| C[type=manifest.artifactType]
+    B -->|No| E[type=manifest.config.mediaType]
+```
+
+For aritifact authors, the the choice of whether to set the `artifactType` or
+`config.mediaType` depends on the contents and shape of the artifact and is discussed in the following sections.
+
+## Dissecting the Manifest
+
+The structure of the artifact depends on the prupose of the artifact and content.
+The manifest is a JSON document that contains the metadata for the artifact.
+Artifacts may have files that are stored as blobs or metadata that is stored in the manifest
+as annotations. The manifest may also contain references to other artifacts.
+
+### Artifacts with blobs
+
+Let's delve into an OCI artifact that uses an [Image Manifest][image-manifest].
+Below is an example where artifact type is `application/vnd.example+type`. The artifact
+also has blobs that are referenced in the manifest in the `layers` property. It is
+important to note that the `config` property is required in the manifest and
+in this case the config is an empty blob as per the [empty descriptor guidance][empty-descriptor].
+
+```json
+ {
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.example+type",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2,
+    "data": "e30="
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar",
+      "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+      "size": 12,
+      "annotations": {
+        "org.opencontainers.image.title": "hello.txt"
+      }
+    }
+  ],
+  "annotations": {
+    "org.opencontainers.image.created": "2023-08-03T00:21:51Z"
+  }
+}
+```
+
+### Artifacts with config
+
+Clients can use `config.mediaType` property  to
```

Comment:
Since these docs are for new implementations, can we change this to discuss `config.mediaType` as a backwards compact scenario?
For instance, artifacts that have a config object, would still use the `artifactType` property. This allows them to decouple the config mediaType from the artifact type. 

Here's a generic example
- `config.mediaType: "application/vnd.org.project.config.v1+yaml"`
- `artifactType: "application/vnd.org.project"`

or, here's what I would propose for helm: (cc @sabre1041, @scottrigby) 

- `config.mediaType: "application/vnd.cncf.helm.config.v1+json"`
- `artifactType: "application/vnd.cncf.helm"`

For instance, a project could reuse a `config.mediaType` from another project. They might want to reuse the container image `config.mediaType`

- `config.mediaType: "application/vnd.oci.image.config.v1+json"`
- `artifactType: "application/vnd.org.project"`

Because the manifest has the `artifactType`, any consumer MUST defer to that to infer it's "type"

Reply:
This is exactly why this spec is here. If there is a config type then that is a valid use case and there is no reason to duplicate the artifact type and config type.

Reply:
The decision tree is fairly complex. See notes below, with suggested updated text. 
If authors or consumers need to parse that decision tree, we won't see much adoption. **_Note_**, this should be documented under https://github.com/opencontainers/image-spec/blob/main/artifacts-guidance.md
But, if we're explaining it here, we really need to find a simpler explanation.

Reply:
Its fair to have the decision tree. This was added after a question from @dasiths